kinesis/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/sinks/KinesisSink.scala

```diff
@@ -302,7 +346,7 @@ class KinesisSink private (
       multiPut(streamName, batch).onComplete {
         case Success(s) => {
           val results = s.getRecords.asScala.toList
-          val failurePairs = batch zip results filter { _._2.getErrorMessage != null }
+          val failurePairs = batch.zip(results).filter(_._2.getErrorMessage != null)
```

Comment:
Yes, the unlimited number of retries is because we keep failed inserts to Kinesis in memory to retry again later. So, we have two levels of retries now. One is baked in Kinesis client, the other is the old mechanism that we always had.

Reply:
So if SQS buffer is activated, the first level of retries (in Kinesis client) still occurs, but not the second, right?